design/general-progress-monitoring.md

```diff
@@ -425,36 +473,41 @@ progress.
 ## Backup workflow changes
 
 The backup workflow remains the same until we get to the point where the `velero-backup.json` object
-is written.  At this point, we will queue the backup to a finalization go-routine.  The next backup
-may then begin.  The finalization routine will run across all of the
-VolumeSnapshotter/BackupItemAction operations and call the _Progress_ method on each of them.
+is written.  At this point, Velero will
+run across all of the VolumeSnapshotter/BackupItemAction operations and call the _Progress_ method
+on each of them.
 
-If all snapshot and backup item operations have finished (either successfully or failed), the backup
-will be completed and the backup will move to the appropriate terminal phase and upload the
-`velero-backup.json` object to the object store and the backup will be complete.
+If all backup item operations have finished (either successfully or failed), the backup will move to
+one of the finalize phases.
 
 If any of the snapshots or backup items are still being processed, the phase of the backup will be
 set to the appropriate phase (_WaitingForPluginOperations_ or
-_WaitingForPluginOperationsPartiallyFailed_).  In the event of any of the progress checks return an
-error, the phase will move to _WaitingForPluginOperationsPartiallyFailed_.  The backup will then be
-requeued and will be rechecked again after some time has passed.
+_WaitingForPluginOperationsPartiallyFailed_), and the async backup operations controller will
```

Comment:
Looking at it now, I think the actual words used in the API leave off async (i.e. command flags, status fields) --  those tend to be very long already, so it makes them more unwieldy. We tend to add "async" in the text descriptions, documentation, etc. to make it clearer though. I could remove "async" from the other part of the messaging, but I actually think they help to make it clearer the way it is now. I don't have a strong opinion here, so we can deal with any changes in a follow-on PR next week. I think for the controller naming, we can leave that as-is either way, since there's nothing user-visible there.

Reply:
Yes, it's trasnparent to the user, I'm just thinking making the naming more consistent and simplified would help the communication between developers, and make the code easier to understand.

Reply:
@reasonerjt if we need everything to either have "async" or not have it in the name, then I think we'd better remove it, since some of the names will be too long and unwieldy with them. I'm ok with removing Async where it appears, if that's ok with everyone. This would, for example, mean that `asyncBackupOperationsReconciler` would become `backupOperationsReconciler`.